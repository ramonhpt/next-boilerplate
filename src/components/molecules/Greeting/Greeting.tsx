'use client'

import { motion } from 'framer-motion'

export function Gretting() {
  return (
    <motion.div
      className="mt-32 flex h-32 w-32 flex-col items-center justify-center rounded-xl bg-gradient-to-r from-purple-400 via-pink-500 to-red-500 p-8"
      initial={{ opacity: 0, scale: 0.9 }}
      animate={{ opacity: 1, scale: 1 }}
      whileHover={{ scale: 1.2, rotate: 90 }}
      whileTap={{
        scale: 0.8,
        rotate: -90,
        borderRadius: '100%'
      }}
    >
      <motion.h1>HIRO</motion.h1>
      <motion.h6>backoffice</motion.h6>
    </motion.div>
  )
}
