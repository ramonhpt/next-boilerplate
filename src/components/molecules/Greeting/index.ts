import { Gretting } from './Greeting'

export default Gretting
