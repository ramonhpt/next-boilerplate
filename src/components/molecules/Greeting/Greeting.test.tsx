import React from 'react'
import { render } from '@testing-library/react'
import { Gretting } from './Greeting'

describe('Greeting Component', () => {
  it('should render', () => {
    render(<Gretting />)
  })

  it('should match snapshot', () => {
    const { container } = render(<Gretting />)
    expect(container).toMatchSnapshot()
  })
})
