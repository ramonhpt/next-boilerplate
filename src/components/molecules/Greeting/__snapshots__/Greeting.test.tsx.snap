// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting Component should match snapshot 1`] = `
<div>
  <div
    class="mt-32 flex h-32 w-32 flex-col items-center justify-center rounded-xl bg-gradient-to-r from-purple-400 via-pink-500 to-red-500 p-8"
    style="opacity: 0; transform: scale(0.9);"
    tabindex="0"
  >
    <h1>
      HIRO
    </h1>
    <h6>
      backoffice
    </h6>
  </div>
</div>
`;
